docs/src/inverse_problems/optimization_ode_param_fitting.md

```diff
@@ -0,0 +1,173 @@
+    kP, SE --> P + E
+end
+```
+From some known initial condition, and a true parameter set (which we later want to recover from the data) we generate synthetic data (on which we will demonstrate the fitting process).
+```@example diffeq_param_estim_1 
+# Define initial conditions and parameters.
+u0 = [:S => 1.0, :E => 1.0, :SE => 0.0, :P => 0.0]
+p_true = [:kB => 1.0, :kD => 0.1, :kP => 0.5]
+
+# Generate synthetic data.
+using OrdinaryDiffEq
+oprob_true = ODEProblem(rn, u0, (0.0, 10.0), p_true)
+true_sol = solve(oprob_true, Tsit5())
+data_sol = solve(oprob_true, Tsit5(); saveat=1.0)
+data_ts = data_sol.t[2:end]
+data_vals = (0.8 .+ 0.4*rand(10)) .* data_sol[:P][2:end]
+
+# Plots the true solutions and the (synthetic data) measurements.
+using Plots
+plot(true_sol; idxs=:P, label="True solution", lw=8)
+plot!(data_ts, data_vals; label="Measurements", seriestype=:scatter, ms=6, color=:blue)
+```
+
+Next, we will use DiffEqParamEstim.jl to build a loss function that describing how well our model's simulations fit the data.
+```@example diffeq_param_estim_1 
+using DiffEqParamEstim, Optimization
+p_dummy = [:kB => 0.0, :kD => 0.0, :kP => 0.0]
+oprob = ODEProblem(rn, u0, (0.0, 10.0), p_dummy)
+loss_function = build_loss_objective(oprob, Tsit5(), L2Loss(data_ts, data_vals), Optimization.AutoForwardDiff(); maxiters=10000, verbose=false, save_idxs=4)
+nothing # hide
+```
+To `build_loss_objective` we provide the following arguments:
+- `oprob`: The `ODEProblem` with which we simulate our model (using some dummy parameter values, since we do not know these).
+- `Tsit5()`: The numeric integrator we wish to simulate our model with.
+- `L2Loss(data_ts, data_vals)`: Defines the loss function. While [other alternatives](https://docs.sciml.ai/DiffEqParamEstim/stable/getting_started/#Alternative-Cost-Functions-for-Increased-Robustness) are available, `L2Loss` is the simplest one (measuring the sum of squared distances between model simulations and data measurements). Its first argument is the time points at which the data is collected, and the second is the data's values.
+- `Optimization.AutoForwardDiff()`: Our choice of [automatic differentiation](https://en.wikipedia.org/wiki/Automatic_differentiation) framework.
+
+Furthermore, we can pass any number of additional optional arguments, these are then passed to the internal `solve()` function (that are used to simulate our ODE). Here we provide the following additional arguments:
+- `maxiters=10000`: If the ODE integrator takes a very large number of steps, that is typically the sign of a very poor fit. Reducing the `maxiters` threshold reduces the time we waste on evaluating such models. 
+- `verbose=false`: The simulation of models with highly unsuitable parameter sets typically generate various warnings (such as simulation terminations due to reaching the `maxiters` value). To avoid an overflow of such (here unnecessary) warnings as we evaluate a large number of parameter sets, we turn warnings off.
```

Comment:
```suggestion
- `verbose=false`: The simulation of models with highly unsuitable parameter sets typically generate various warnings (such as premature simulation termination due to reaching `maxiters` timesteps). To avoid an overflow of such (here unnecessary) warnings, as we evaluate a large number of parameter sets, we turn warnings off.
```